packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -607,6 +610,7 @@ export function createSdkContext<
     __namespaceToApiVersionParameter: new Map(),
     __tspTypeToApiVersions: new Map(),
     __namespaceToApiVersionClientDefaultValue: new Map(),
+    decoratorsWhiteList: options?.decoratorsWhiteList ?? defaultDecoratorsWhiteList,
```

Comment:
I think the config coming from sdk should be in addition to, not a replacement of.  We could call it something that indicates this like `additionalDecorators`